File: rust/src/pcl.rs
use std::env;
use std::io::{self, Write, stderr, stdin};
use std::process;

use pol::{pcl, pescr, plleg};

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() != 3 {
        writeln!(stderr(), "./pcl alf bet").unwrap();
        process::exit(-1);
    }

    let alf: f64 = match args[1].parse() {
        Ok(val) => val,
        Err(_) => {
            writeln!(stderr(), "./pcl alf bet").unwrap();
            process::exit(-1);
        }
    };

    let bet: f64 = match args[2].parse() {
        Ok(val) => val,
        Err(_) => {
            writeln!(stderr(), "./pcl alf bet").unwrap();
            process::exit(-1);
        }
    };

    let (n1, p1) = match plleg() {
        Ok(result) => result,
        Err(_) => {
            writeln!(stderr(), "pcl: error llegint el primer polinomi!").unwrap();
            process::exit(-2);
        }
    };

    let (n2, p2) = match plleg(&mut stdin()) {
        Ok(result) => result,
        Err(_) => {
            writeln!(stderr(), "pcl: error llegint el segon polinomi!").unwrap();
            process::exit(-2);
        }
    };

    let n3 = if n2 >= n1 { n2 } else { n1 };
    let mut p3 = vec![0.0; n3 + 1];

    pcl(n1, &p1, n2, &p2, alf, bet, &mut p3);
    pescr(n3, &p3, &mut io::stdout());

    process::exit(0);
}

File: pcl.c
#include <stdio.h>
#include <stdlib.h>
#include "pol.h"

int main (int argc, char *argv[]) {
    int n1, n2, n3;
    double alf, bet, *p1, *p2, *p3;

    if (argc!=3
         || sscanf(argv[1],"%lf",&alf)!=1
         || sscanf(argv[2],"%lf",&bet)!=1) {
        fprintf(stderr, "./pcl alf bet\n");
        return -1;
    }

    if (plleg(stdin,&n1,&p1)) {
        fprintf(stderr, "pcl: error llegint el primer polinomi!\n");
        return -2;
    }
    if (plleg(stdin,&n2,&p2)) {
        fprintf(stderr, "pcl: error llegint el segon polinomi!\n");
        free(p1);
        return -2;
    }

    n3=n1; if (n2>=n3) n3=n2;
    p3=malloc((1+n3)*sizeof(double));
    if (p3==NULL) {
        fprintf(stderr, "Error reservant memòria!\n");
        free(p1); free(p2);
        return -4;
    }
    pcl(n1,p1,n2,p2,alf,bet,p3);
    pescr(n3,p3,stdout);

    free(p1);
    free(p2);
    free(p3);
    return 0;
}
